src-docs/src/views/side_nav/side_nav.js

```diff
@@ -1,67 +1,57 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 
 import { EuiSideNav } from '../../../../src/components';
 
-export default class extends Component {
-  constructor(props) {
-    super(props);
+export default function() {
+  const [isSideNavOpenOnMobile, setisSideNavOpenOnMobile] = useState(false);
 
-    this.state = {
-      isSideNavOpenOnMobile: false,
-    };
-  }
-
-  toggleOpenOnMobile = () => {
-    this.setState({
-      isSideNavOpenOnMobile: !this.state.isSideNavOpenOnMobile,
-    });
+  const toggleOpenOnMobile = () => {
+    setisSideNavOpenOnMobile(!this.state.isSideNavOpenOnMobile);
```

Comment:
It's not working on mobile because, in a function component, we have no `this`, so we can’t assign or read `this.state`.

```suggestion
    setisSideNavOpenOnMobile(!isSideNavOpenOnMobile);
```